com.microsoft.mrtk.input/Interactors/XRRayInteractorExtensions.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+	/// <summary>
+	/// Extensions to the XRRayInteractor and associated structs.
+	/// </summary>
+	public static class XRRayInteractorExtensions
+	{
+		/// <summary>
+		/// Used to locate and lock the raycast hit data on a select.
+		/// </summary>
+		/// <param name="rayInteractor">The XRRayInteractor responsible for the raycast hit.</param>
+		/// <param name="interactableObject"> The IXRSelectInteractable which has been selected. </param>
+		/// <returns>The local position and normal of the hit target, the hit target transform, and a reference point to calculate hit distance, contained in a TargetHitInfo struct.</returns>
+		public static TargetHitInfo LocateTargetHitPoint(this XRRayInteractor rayInteractor, IXRSelectInteractable interactableObject)
```

Comment:
Sorry @marlenaklein-msft ,   i have another suggestion :)

Someone using this function in the future may want to know if there was an actual "hit", so can you change the function signature to 

`public static bool TryGetHitDetails(this XRRayInteractor rayInteractor, IXRSelectInteractable interactableObject, out TargetHitDetails details)`

And return `true` if `TryGetCurrentRaycast` succeeds, and `false` otherwise.

The current consumers of the function should ignore the Boolean, because of the `ISnapInteractable` and `IScrollable` logic.  However future users of the function will probably want to know if `TryGetCurrentRaycast` succeeded.

Also, I think we should avoid the "info" abbreviation, even if its commonly used.  Can we change `TargetHitInfo` to `TargetHitDetails`.